cargo-concordium/src/build.rs

```diff
@@ -332,16 +350,34 @@ pub fn build_contract_schema<A>(
     cargo_args: &[String],
     generate_schema: impl FnOnce(&[u8]) -> ExecResult<A>,
 ) -> anyhow::Result<A> {
-    let metadata = MetadataCommand::new()
-        .exec()
-        .context("Could not access cargo metadata.")?;
+    let mut args = std::env::args().skip_while(|val| !val.starts_with("--manifest-path"));
+
+    let mut cmd = MetadataCommand::new();
+    match args.next() {
+        Some(ref p) if p == "--manifest-path" => {
+            // If a `--manifest-path` is provided use this path as the package root to
+            // build the schema.
+            cmd.manifest_path(args.next().context(
+                "The argument '--manifest-path <manifest-path>' requires a value but none was \
+                 supplied.",
+            )?);
+        }
+        Some(p) => {
+            // If a `--manifest-path` is provided use this path as the package root to
+            // build the schema.
+            cmd.manifest_path(p.trim_start_matches("--manifest-path="));
+        }
+        None => {}
+    };
+
+    let metadata = cmd.exec().context("Could not access cargo metadata.")?;
```

Comment:
Could we please factor this into a function
```rust
fn get_crate_metadata(extra_args: &[String]) -> anyhow::Result<Metadata> {
    let mut args = extra_args.iter().skip_while(|val| !val.starts_with("--manifest-path"));
    let mut cmd = MetadataCommand::new();
    match args.next() {
        Some(ref p) if p == "--manifest-path" => {
            // If a `--manifest-path` is provided use this path as the package root to
            // build the schema.
            cmd.manifest_path(args.next().context(
                "The argument '--manifest-path <manifest-path>' requires a value but none was \
                 supplied.",
            )?);
        }
        Some(p) => {
            // If a `--manifest-path` is provided use this path as the package root to
            // build the schema.
            cmd.manifest_path(p.trim_start_matches("--manifest-path="));
        }
        None => {}
    };

    cmd.exec().context("Could not access cargo metadata.")
}
```
Since it is repeated multiple places.